### ISSUE TITLE
TWAP: Make integration tests for relevant methods working on 3 asset pools

### ISSUE THREAD

**Issue by ValarDragon** [osmosis-labs/osmosis#2208]
## Background

Most of the TWAP tests, are centered around 2 asset pools. All of the store management and code is written to explicitly work with more pairs, and _should_ work, with a few unit tests for this.

We should make multi-asset pool tests at the api_test and hook_test layers.

## Acceptance Criteria

- [x] Multi asset pool test for Endblock flow
- [x] Multi asset pool test for `GetArithmeticTwap`

**Comment by mattverse** [comment]
Blocked on https://github.com/osmosis-labs/osmosis/pull/2340

**Comment by mattverse** [comment]
- [ ] https://github.com/osmosis-labs/osmosis/pull/2364

**Comment by czarcas7ic** [comment]
Closing in favor of #2538
